e2e-tests/browser/openid4vp_employeecredential/main_test.go

```diff
@@ -0,0 +1,123 @@
+//go:build e2e_tests
+
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package openid4vp_employeecredential
+
+import (
+	"github.com/chromedp/chromedp"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/e2e-tests/browser"
+	iamAPI "github.com/nuts-foundation/nuts-node/e2e-tests/browser/client/iam"
+	"github.com/nuts-foundation/nuts-node/e2e-tests/browser/rfc019_selfsigned/apps"
+	didAPI "github.com/nuts-foundation/nuts-node/vdr/api/v2"
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+	"time"
+)
+
+var nodeClientConfig = core.ClientConfig{Address: "http://localhost:8081"}
+
+func init() {
+	//os.Setenv("SHOW_BROWSER", "true")
+	//os.Setenv("KEEP_BROWSER_OPEN", "true")
+}
+
+func Test_UserAccessToken_EmployeeCredential(t *testing.T) {
```

Comment:
Note: Definitely shorter then our other e2e tests, but also less of an example/test on raw http requests. (We can make the same mistake in client/server which still make the test succeed).

Reply:
We could opt to use raw(ish) HTTP requests in our e2e tests, which would definitely improve effectiveness (then the chances of a bug in the API and client will have much more chance of being found) and it would be more documentation on how the API is supposed to be used. At the cost of maintainability.

But as the number of e2e tests we have is limited, I think it's OK. But we'll probably make a mini-client in the e2e tests because you don't want to have the "create DID" HTTP request and response handling over and over again.

Reply:
I'd say have that refactor separate from this feature change. So merge this first, then clean up?